tests/tests_pytorch/tuner/test_scale_batch_size.py

```diff
@@ -140,6 +140,31 @@ def test_auto_scale_batch_size_trainer_arg(tmpdir, scale_arg):
     assert not os.path.exists(tmpdir / "scale_batch_size_temp_model.ckpt")
 
 
+@RunIf(min_cuda_gpus=1)
+@pytest.mark.parametrize("scale_mode", ["power", "binsearch"])
+def test_auto_scale_batch_size_calls_forward_method_x_times(tmpdir, scale_mode):
+    """Test that models actually go through forward loop (and don't get skipped due to buggy epoch count
+    resets)."""
+
+    class CounterModel(BatchSizeModel):
+        def __init__(self):
+            super().__init__(batch_size=2)
+            self.num_completed_epochs = 0
+
+        def on_train_epoch_end(self):
+            self.num_completed_epochs += 1
+
+    tutils.reset_seed()
+    model = CounterModel()
+    trainer = Trainer(default_root_dir=tmpdir, auto_scale_batch_size=True, max_epochs=1, accelerator="gpu", devices=1)
+    max_trials = 5
+    trainer.tune(model, scale_batch_size_kwargs={"max_trials": max_trials, "mode": scale_mode})
+
+    expected_minimum_number_of_passed_epochs = max_trials
+    actual_number_of_passed_epochs = model.num_completed_epochs
+    assert actual_number_of_passed_epochs >= expected_minimum_number_of_passed_epochs
+
+
```

Comment:
I think this test can be integrated with the last one (test_dataloader_reset_with_scale_batch_size) where you can Mock the calls of `on_train_epoch_end`.

Reply:
Sure, I can do that, but I would argue that it is nicer to have them separated, since currently each of these tests has a dedicated purpose?

Reply:
yeah, but every `trainer.tune` call increases the CI time and it's fine to merge certain assertions if one of them is not doing any significant work.